src/java_binding/src/iterator.rs

```diff
@@ -0,0 +1,191 @@
+use risingwave_storage::hummock::local_version::pinned_version::PinnedVersion;
+use risingwave_storage::hummock::shared_buffer::shared_buffer_batch::SharedBufferBatch;
+use risingwave_storage::hummock::store::state_store::HummockStorageIterator;
+use risingwave_storage::hummock::store::version::HummockVersionReader;
+use risingwave_storage::hummock::{SstableStore, TieredCache};
+use risingwave_storage::monitor::StateStoreMetrics;
+use risingwave_storage::storage_value::StorageValue;
+use risingwave_storage::store::{ReadOptions, StreamTypeOfIter};
+use tokio::runtime::Runtime;
+use tokio::sync::mpsc::unbounded_channel;
+
+pub struct Iterator {
+    runtime: Runtime,
+    row_serializer: RowDeserializer,
+    stream: Pin<Box<StreamTypeOfIter<HummockStorageIterator>>>,
+}
+
+pub struct Record {
+    key: Vec<u8>,
+    row: OwnedRow,
+}
+
+impl Record {
+    pub fn key(&self) -> &[u8] {
+        self.key.as_slice()
+    }
+
+    pub fn is_null(&self, idx: usize) -> bool {
+        self.row[idx].is_some()
+    }
+
+    pub fn get_int64(&self, idx: usize) -> i64 {
+        match self.row[idx].as_ref().unwrap() {
+            ScalarImpl::Int64(num) => *num,
+            _ => unreachable!("type is not int64 at index: {}", idx),
+        }
+    }
+
+    pub fn get_utf8(&self, idx: usize) -> &str {
+        match self.row[idx].as_ref().unwrap() {
+            ScalarImpl::Utf8(s) => s.as_ref(),
+            _ => unreachable!("type is not utf8 at index: {}", idx),
+        }
+    }
+}
+
+const TEST_EPOCH: u64 = 1000;
+const TEST_TABLE_ID: TableId = TableId { table_id: 2333 };
+
+fn gen_mock_schema() -> Vec<DataType> {
+    vec![DataType::Int64, DataType::Varchar, DataType::Int64]
+}
+
+fn gen_mock_imm_lists() -> Vec<SharedBufferBatch> {
+    let rows = vec![
+        OwnedRow::new(vec![
+            Some(ScalarImpl::Int64(100)),
+            Some(ScalarImpl::Utf8("value_of_100".to_owned_scalar())),
+            None,
+        ]),
+        OwnedRow::new(vec![
+            Some(ScalarImpl::Int64(101)),
+            Some(ScalarImpl::Utf8("value_of_101".to_owned_scalar())),
+            Some(ScalarImpl::Int64(2333)),
+        ]),
+    ];
+    let data_chunk = DataChunk::from_rows(&rows, &gen_mock_schema());
+    let row_data = data_chunk.serialize();
+    let kv_pairs = row_data
+        .into_iter()
+        .enumerate()
+        .map(|(i, row)| {
+            (
+                Bytes::from(format!("key{:?}", i)),
+                StorageValue::new_put(row),
+            )
+        })
+        .collect_vec();
+
+    let sorted_items = SharedBufferBatch::build_shared_buffer_item_batches(kv_pairs);
+    let size = SharedBufferBatch::measure_batch_size(&sorted_items);
+    let imm = SharedBufferBatch::build_shared_buffer_batch(
+        TEST_EPOCH,
+        sorted_items,
+        size,
+        vec![],
+        TEST_TABLE_ID,
+        None,
+    );
+
+    vec![imm]
+}
+
+impl Iterator {
+    pub fn new() -> StorageResult<Self> {
```

Comment:
It looks like a demo or test?

Reply:
Yes. This PR is only for defining and testing the JNI binding interface between Java and rust, so the data and schema are predefined. In a later PR, we will read from catalog and pass the table schema and hummock version from java to rust when we create a new iterator.